Src/SDK/Common/Temporal.Common.Payloads/public/PayloadContainers.Unnamed.Empty.cs

```diff
@@ -25,7 +25,8 @@ public TVal GetValue<TVal>(int index)
 
                 public bool TryGetValue<TVal>(int index, out TVal value)
                 {
-                    throw CreateNoSuchIndexException(index);
+                    value = default(TVal);
```

Comment:
I made this change because try methods shouldn't throw except in the exceptional cases. An index out of bounds is the case we are handling here, so that is why I thought it shouldn't throw. However, there is another class where I left the exception in TryGetValue. It was because type deserialization fails. I also don't think that method should throw an exception, but without a huge refactor to make TryGetValue return why it didn't work, I think throwing an exception is the best way to go.

Reply:
agreed. :)

btw,. where did you need to leave the exception?

Reply:
I left the exception in PayloadContainers.Unnamed.SerialiazedDataBacked.TryGetValue 
https://github.com/mightyshazam/sdk-dotnet/blob/9da8848d6c70e056c7c1f9126240f48672164f31/Src/SDK/Common/Temporal.Common.Payloads/public/PayloadContainers.Unnamed.SerializedDataBacked.cs#L105
I was conflicted since I don't think Try methods should throw, but it is throwing for a reason different than all other methods.

Reply:
ToDo: Probably need to make it return false. :)

Reply:
I thought about it and left the exception. To clarify, I added the following code comment:

```
// The "Try"-pattern contract of this API implies no exceptions. However, so does the 
// downstream call to `TryDeserialize`. We trust the downstream call: It should just
// return true/false. However, if something truly exceptional happens and it throws,
// we enrich that exception a clarifying message and let it bubble up.
```